pennylane/ops/op_math/controlled_decompositions.py

```diff
@@ -458,14 +458,94 @@ def ctrl_decomp_bisect(
 
 
 def decompose_mcx(control_wires, target_wire, work_wires):
-    """Decomposes the multi-controlled PauliX gate"""
-
+    """Decomposes the multi-controlled PauliX gate using decompositions from
+    `Barenco et al. (1995) <https://arxiv.org/abs/quant-ph/9503016>`_"""
     num_work_wires_needed = len(control_wires) - 2
 
+    if len(control_wires) == 1:
+        return [qml.CNOT(wires=control_wires + Wires(target_wire))]
+    if len(control_wires) == 2:
+        return qml.Toffoli.compute_decomposition(wires=control_wires + Wires(target_wire))
+
     if len(work_wires) >= num_work_wires_needed:
+        # Lemma 7.2
         return _decompose_mcx_with_many_workers(control_wires, target_wire, work_wires)
+    if len(work_wires) >= 1:
+        # Lemma 7.3
+        return _decompose_mcx_with_one_worker(control_wires, target_wire, work_wires[0])
+
+    # Lemma 7.5
+    with qml.QueuingManager.stop_recording():
+        op = qml.X(target_wire)
+    return _decompose_multicontrolled_unitary(op, control_wires)
+
+
+def _decompose_multicontrolled_unitary(op, control_wires):
+    """Decomposes general multi controlled unitary with no work wires
+    Follows approach from Lemma 7.5 combined with 7.3 and 7.2 of
+    https://arxiv.org/abs/quant-ph/9503016.
+
+    We are assuming this decomposition is used only in the general cases
+    """
+    # pylint: disable=import-outside-toplevel
+    from pennylane.ops.op_math.controlled import _is_single_qubit_special_unitary
+
+    if not op.has_matrix or len(op.wires) != 1:
+        raise ValueError(
+            "The target operation must be a single-qubit operation with a matrix representation"
+        )
+
+    target_wire = op.wires
+    if len(control_wires) == 0:
+        return [op]
+    if len(control_wires) == 1:
+        return ctrl_decomp_zyz(op, control_wires)
+    if _is_single_qubit_special_unitary(op):
+        return ctrl_decomp_bisect(op, control_wires)
+    # use recursive decomposition of general gate
+    return _decompose_recursive(op, 1.0, control_wires, target_wire, Wires([]))
+
 
-    return _decompose_mcx_with_one_worker(control_wires, target_wire, work_wires[0])
+def _decompose_recursive(op, power, control_wires, target_wire, work_wires):
+    """Decompose multicontrolled operator recursively using lemma 7.5
+    Number of gates in decomposition are: O(len(control_wires)^2)
+    """
+    if len(control_wires) == 1:
+        with qml.QueuingManager.stop_recording():
+            powered_op = qml.pow(op, power)
```

Comment:
Minor nitpick that can be ignored.

For these `pow` operations, we will likely be taking their matrix later. And then using the matrix to calculate the rot angles.

Even if these are just 2x2, taking a fractional matrix power of an operator isn't trivial. I think we should consider using `qml.pow(op, power, lazy=False)` here.  That would do such things as turning `qml.X(0) ** 0.5` into a `qml.SX(0)`, or `qml.RX(1.5, 0) ** 0.5` into `qml.RX(0.5 * 1.5, 0)`.  

This comment applies to the below lines using `qml.pow` as well.  

This comment may be optimizing for things that don't end up mattering in the grand scheme of things.

<img width="1181" alt="Screenshot 2024-06-05 at 2 42 33 PM" src="https://github.com/PennyLaneAI/pennylane/assets/6364575/347330f1-794d-4166-8f39-e02aee2795f1">